scripts/workspace_cromwell.py

```diff
@@ -0,0 +1,294 @@
+import os
+import json
+from pathlib import Path
+import requests
+import subprocess
+import sys
+import time
+import uuid
+
+
+# If a CROMWELL app exists return (app_name, status), otherwise (None, None).
+def check_for_app(env):
+    list_apps_url = f'{env["leonardo_url"]}/api/google/v1/apps/{env["google_project"]}'
+    r = requests.get(
+        list_apps_url,
+        params={
+          'includeDeleted': 'false'
+        },
+        headers = {
+            'Authorization': f'Bearer {env["token"]}'
+        }
+    )
+    r.raise_for_status()
+
+    for potential_app in r.json():
+        if potential_app['appType'] == 'CROMWELL' and (
+                str(potential_app['auditInfo']['creator']) == env['owner_email']
+                or str(potential_app['auditInfo']['creator']) == env['user_email']
+        ) :
+            potential_app_name = potential_app['appName']
+            potential_app_status = potential_app['status']
+
+            # We found a CROMWELL app in the correct google project and owned by the user. Now just check the workspace:
+            _, workspace_namespace, workspace_name, _ = get_app_details(env, potential_app_name)
+            if workspace_namespace == env['workspace_namespace'] and workspace_name == env['workspace_name']:
+                return potential_app_name, potential_app_status
+
+    return None, None
+
+
+# Returns (status, workspace_namespace, workspace_name, proxyUrls) for a running app, (None, None, None, None) for a missing/shutdown one:
+def get_app_details(env, app_name):
+    get_app_url = f'{env["leonardo_url"]}/api/google/v1/apps/{env["google_project"]}/{app_name}'
+    r = requests.get(
+        get_app_url,
+        params={
+            'includeDeleted': 'true'
+        },
+        headers={
+            'Authorization': f'Bearer {env["token"]}'
+        }
+    )
+    if r.status_code == 404:
+        return 'DELETED', None, None, None
+    else:
+        r.raise_for_status()
+    result_json = r.json()
+    custom_environment_variables = result_json['customEnvironmentVariables']
+    return result_json['status'], custom_environment_variables['WORKSPACE_NAMESPACE'], custom_environment_variables['WORKSPACE_NAME'], result_json.get('proxyUrls')
+
+
+# If a Cromwell PD exists, return pd_name. Otherwise None.
+def check_for_pd(env):
+    list_apps_url = f'{env["leonardo_url"]}/api/google/v1/disks/{env["google_project"]}'
+    r = requests.get(
+        list_apps_url,
+        params={
+            'saturnApplication': 'CROMWELL',
+            'includeDeleted': 'false'
+        },
+        headers={
+            'Authorization': f'Bearer {env["token"]}'
+        }
+    )
+    r.raise_for_status()
+
+    for potential_disk in r.json():
```

Comment:
Should we check whether this disk was created by user or not here? Because it seems I might have access to disks that are created by other users in the workspace similar to list apps api

Reply:
Good catch! Added the user check here too